pkg/controller/storagecluster/cephcluster_test.go

```diff
@@ -350,3 +353,75 @@ func TestStorageClassDeviceSetCreation(t *testing.T) {
 	}
 
 }
+
+func createDummyKMSConfigMap(kmsProvider, kmsAddr string) *corev1.ConfigMap {
+	cm := &corev1.ConfigMap{}
+	cm.Name = KMSConfigMapName
+	cm.Data = make(map[string]string)
+	cm.Data["KMS_PROVIDER"] = kmsProvider
+	cm.Data["VAULT_ADDR"] = kmsAddr
+	cm.Data["VAULT_BACKEND_PATH"] = "ocs"
+	cm.Data["VAULT_NAMESPACE"] = "my-ocs-namespace"
+	return cm
+}
+
+func TestKMSConfigChanges(t *testing.T) {
```

Comment:
Please refactor this test to use case tables.

Reply:
Done

Reply:
The general idea is there, but the format is not ideal. There should be a struct of parameters, followed by a list of parameters for individual test cases, and then one for loop to run the same exact test over all values.

Reply:
Updated the code...

Reply:
You still have two separate loops. They seem like they could be one loop with a small if statement.

Reply:
Also, the test parameters usually take a case number with a descriptor of what the case is testing

Reply:
> You still have two separate loops. They seem like they could be one loop with a small if statement.

But isn't a better idea to add the valid cases and invalid cases separate, rather than cloging the code with if conditions (where even the valid cases has to go through the condition, just to fail). This way it is easier, in the future, to add more new cases (as the code changes)

> Also, the test parameters usually take a case number with a descriptor of what the case is testing

Checking on this...

Reply:
Merging the loops would give me better confidence that the tests are testing the same code. In this specific instance it's not a problem, but I would like to promote the overall pattern.

Reply:
@jarrpa , I don't see a point in adding a case label just for the sake of it. The point I am trying to make here is the following,

- a. These labels are not used any where else except in the top struct array elments. We could have even used comments instead

- b. We could have easily generated this case label from loop index itself, why add a static text

- c. Case labels would have been much useful if we were adding a description of what test-scenario we are performing with the following parameters.

Nevertheless I have added the labels (and tried to use them) inorder to make the tests compatible with other tests.

Reply:
> Merging the loops would give me better confidence that the tests are testing the same code. In this specific instance it's not a problem, but I would like to promote the overall pattern.

Sorry @jarrpa , saw this comment after the initial push. Let me know if you still want this change...

Reply:
>     * c. Case labels would have been much useful if we were adding a description of what test-scenario we are performing with the following parameters.

There's actually something we can do to take advantage of that, but I'm waiting for the feature freeze and SDK overhaul before bringing it in.

> > Merging the loops would give me better confidence that the tests are testing the same code. In this specific instance it's not a problem, but I would like to promote the overall pattern.
> 
> Sorry @jarrpa , saw this comment after the initial push. Let me know if you still want this change...

Yes please. :)

Reply:
Done...